rgf/utils.py

```diff
@@ -212,8 +212,7 @@ def sparse_savetxt(filename, input_array, including_header=True):
     zip_func = six.moves.zip
     if sp.isspmatrix_csr(input_array):
         input_array = input_array.tocoo()
-    elif not sp.isspmatrix_coo(input_array):
-        input_array = input_array.tocsr().tocoo()
+    input_array = input_array.tocsr().tocoo()
```

Comment:
if input_array is csr, 
Once `input_array.tocoo()` and then `input_array.tocsr().tocoo()`.

Is it redundant?
Do you mean
```python
    if sp.isspmatrix_csr(input_array):
        input_array = input_array.tocoo()
    else:
        input_array = input_array.tocsr().tocoo()
```
Or
```python
    if not sp.isspmatrix_csr(input_array):
        input_array = input_array.tocsr()
    input_array = input_array.tocoo()
```
?

Reply:
Hm... All tests were made with proposed code.
Of course, I wanted to write the first variant :-)

Thanks for catching a bug! I'll retest again.

Reply:
Even your current code, finally array transform to coo.

But csr->coo->cst->coo transformation if input is csr.
It is redundant, but the test passes